app/views/layouts/_menu.html.erb

```diff
@@ -1,11 +1,32 @@
 <div class="collapse navbar-collapse navbar-responsive-collapse">
   <ul class="nav navbar-nav navbar-right">
-    <% cache_unless(can_administrate?(:manage, Menu), ['main_menu', I18n.locale]) do %>
-      <% global_menus = GlobalMenus.new %>
-      <%= menu_dropdown(menus: global_menus.guild_menus, title: t('model.menus.guild')) %>
-      <%= menu_dropdown(menus: global_menus.member_menus, title: t('model.menus.members')) %>
-      <%= menu_dropdown(menus: global_menus.company_menus, title: t('model.menus.companies')) %>
-      <%= menu_dropdown(menus: global_menus.contact_menus, title: t('model.menus.contact')) %>
+    <% cache(['main_menu', I18n.locale], skip_digest: true) do %>
+      <% MainMenu.includes(:translations).index.each do |m| %>
+        <% if m.mega %>
+          <li class="dropdown menu-large">
+            <a href="#" class="dropdown-toggle" data-toggle="dropdown">
+              <%= safe_join([m, ' ', fa_icon('angle-down')]) %>
+            </a>
+            <ul class="dropdown-menu megamenu row <%= m.fw ? 'fw' : '' %>">
+              <% menu = m.menus.includes(:translations).index %>
+              <% col = menu.select(:column).distinct(:column).count %>
```

Comment:
The whole `col` logic should perhaps be in a helper.
Is the amount of columns limited? 1-4?